### PR TITLE
Add X Drive field oriented control

### PR DESCRIPTION
**Pull request by jazonshou**
### Description of the Change

Add field oriented control to the X Drive model

### Motivation

It did not exist beforehand

### Possible Drawbacks

None that I can think of right now

### Verification Process

Tested on X Drive & it works

### Applicable Issues

n/a

### CODE REVIEW

**Comment by Octogonapus** [comment]
Seems good. Please add tests, though.

File: src/api/chassis/model/xDriveModel.cpp [hunk 1]
```diff
@@ -216,6 +216,39 @@ void XDriveModel::xArcade(const double ixSpeed,
...
+  if (std::abs(forwardSpeed) < ithreshold) {
+    forwardSpeed = 0;
+  }
+
+  double yaw = std::clamp(iyaw, -1.0, 1.0);
+  if (std::abs(yaw) < ithreshold) {
+    yaw = 0;
+  }
+
+  double fwd = forwardSpeed * std::cos(iangle.convert(okapi::radian)) - xSpeed * std::sin(iangle.convert(okapi::radian));
```
Comment (theol0403) [inline src/api/chassis/model/xDriveModel.cpp]: The cos/sin operations can be simplified to `cos(iangle)`.

Comment (jazonshou) [reply src/api/chassis/model/xDriveModel.cpp]: Got it, thanks!

**Comment by jazonshou** [comment]
> Seems good. Please add tests, though.

Thanks, I added some tests

File: include/okapi/api/chassis/model/xDriveModel.hpp [hunk 2]
```diff
@@ -124,6 +125,21 @@ class XDriveModel : public ChassisModel {
   virtual void
   xArcade(double irightSpeed, double iforwardSpeed, double iyaw, double ithreshold = 0);
 
+  /**
+   * Drive the robot with a field-oriented arcade drive layout. Uses voltage mode.
```
Comment (Octogonapus) [inline include/okapi/api/chassis/model/xDriveModel.hpp:129]: Could you please add two one-line examples, one with an angle of zero degrees, and one with an angle of +90 degrees?

File: include/okapi/api/chassis/model/xDriveModel.hpp [hunk 3]
```diff
@@ -124,6 +125,21 @@ class XDriveModel : public ChassisModel {
...
+   * 
+   * @param ixSpeed speed to the right
+   * @param iforwardSpeed speed forward
+   * @param iyaw turn speed
+   * @param iangle current chassis angle
```
Comment (Octogonapus) [inline include/okapi/api/chassis/model/xDriveModel.hpp]: Can you also please document the winding direction of this angle? You just need to state that zero degrees applies no correction and that it winds clockwise.

**Review by Octogonapus — CHANGES_REQUESTED** [review]
Implementation and tests look good. Just two docs comments.

File: include/okapi/api/chassis/model/xDriveModel.hpp (hunk 2 above)
Comment (jazonshou) [reply include/okapi/api/chassis/model/xDriveModel.hpp:129]: Ok, done!

File: include/okapi/api/chassis/model/xDriveModel.hpp (hunk 3 above)
Comment (jazonshou) [reply include/okapi/api/chassis/model/xDriveModel.hpp]: Yep, done!

File: include/okapi/api/chassis/model/xDriveModel.hpp [hunk 4]
```diff
@@ -124,6 +125,27 @@ class XDriveModel : public ChassisModel {
...
+   *    Both fieldOrientedXArcade(1, 0, 0, 0_deg) and fieldOrientedXArcade(1, 0, 0, 90_deg)
+   *    will drive the chassis in the forward/north direction. In other words, no matter
+   *    what the robot's heading is, the robot will move forward/north when you tell it 
+   *    to move forward/north and will move right/east when you tell it to move right/east.
+   *    
+   * 
+   * @param ixSpeed forward speed -- (+1) forward, (-1) backward
+   * @param iySpeed sideways speed -- (+1) right, (-1) left
+   * @param iyaw turn speed -- (+1) clockwise, (-1) counter-clockwise
+   * @param iangle current chassis angle (0_deg = no correction, winds clockwise)
```
Comment (Octogonapus) [inline include/okapi/api/chassis/model/xDriveModel.hpp]:
```suggestion
   *    Both `fieldOrientedXArcade(1, 0, 0, 0_deg)` and `fieldOrientedXArcade(1, 0, 0, 90_deg)`
   *    will drive the chassis in the forward/north direction. In other words, no matter
   *    the robot's heading, the robot will move forward/north when you tell it 
   *    to move forward/north and will move right/east when you tell it to move right/east.
   *    
   * 
   * @param ixSpeed forward speed -- (`+1`) forward, (`-1`) backward
   * @param iySpeed sideways speed -- (`+1`) right, (`-1`) left
   * @param iyaw turn speed -- (`+1`) clockwise, (`-1`) counter-clockwise
   * @param iangle current chassis angle (`0_deg` = no correction, winds clockwise)
```

**Review by Octogonapus — APPROVED** [review]

**Pull request merged**